src/libraries/System.Private.CoreLib/src/System/Text/StringBuilder.cs

```diff
@@ -1054,10 +1054,15 @@ private void AppendWithExpansion(char value)
 
         private StringBuilder AppendSpanFormattable<T>(T value) where T : ISpanFormattable
         {
-            Debug.Assert(typeof(T).Assembly.Equals(typeof(object).Assembly), "Implementation trusts the results of TryFormat because T is expected to be something known");
-
             if (value.TryFormat(RemainingCurrentChunk, out int charsWritten, format: default, provider: null))
             {
+                if ((uint)charsWritten > (uint)RemainingCurrentChunkLength)
```

Comment:
Does this measurably slow down the _existing_ uses of overloads like `Append(int)`?

Reply:
No noticeable impact measured.
See [the new benchmark](https://github.com/dotnet/runtime/pull/92197#issuecomment-1806894975) for more details